components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -132,7 +132,7 @@ BraveSyncClient* BraveSyncServiceImpl::GetSyncClient() {
 }
 
 bool BraveSyncServiceImpl::IsSyncConfigured() {
-  return sync_configured_;
+  return sync_configured_ && (sync_prefs_->GetSyncDevices()->size() >= 2);
```

Comment:
can you elaborate why we need this change? we will reset sync chain when containing only one device https://github.com/brave/brave-core/blob/ede3ca626e86a5d60ba12786b0e74c91870e2acd/components/brave_sync/brave_sync_service_impl.cc#L500

Reply:
@darkdh 
In the situation described in the issue , https://github.com/brave/brave-browser/issues/3190#issuecomment-514802188 , when second `OnSetupSyncHaveCode/OnSetupSyncNewToSync` happens , there is already one device in the devices list, so without patch above execution will not reach `ForceCompleteReset();` in `OnSetupSyncHaveCode/OnSetupSyncNewToSync` .

The more clear way would be to keep `IsSyncConfigured()` as it was and modify `OnSetupSyncHaveCode/OnSetupSyncNewToSync` to 

```
  if (IsSyncConfigured() && (sync_prefs_->GetSyncDevices()->size() >= 2)) {
    NotifyLogMessage("already configured");
    return;
  }
```

Reply:
it is ok to leave it in `IsSyncConfigured`, can you put some comments in it?

Reply:
thanks, fixed at
https://github.com/brave/brave-core/pull/3038/files#diff-c04ecef4235e311e0a383fe4fabcb4d4R135